CorsixTH/Lua/world.lua

```diff
@@ -2253,14 +2253,35 @@ function World:getObject(x, y, id)
   return -- nil
 end
 
+--! Remove all litter from a rectangular area.
+--!param x (int) Start x position of the area.
+--!param y (int) Start y position of the area.
+--!param w (int) Number of tiles in x direction.
+--!param h (int) Number of tiles in y direction.
+function World:removeAllLitterFromArea(x, y, w, h)
+  for xpos = x, x + w - 1 do
+    for ypos = y, y + h - 1 do
+      -- Loop repeatedly over the objects, as there may be more than one litter object in a tile.
+      while true do
+        local litter = self:getObject(xpos, ypos, "litter")
```

Comment:
More efficient to use self:getObjects and loop through them here. Might be useful to extend getObjects to take type filter there instead (then simplify the getObject function).

Not critical.